File: driver.cpp
/*********
 *
 * Driver.cpp
 *
 * This file reads input and then retrieves parsed output
 *
 ********/

#include <iostream>
#include <fstream>
#include "BusinessCardParser.h"

using namespace std;

int main(int argc, char *argv[])
{
    if (argc != 2) {
        cout << "usage: " << argv[0] << " <filename>\n";
        return 1;
    }
    
    // Read from file
    ifstream input;
    input.open(argv[1]);
    if (!input.is_open()) {
        cout << "Unable to read file\n";
        return 2;
    }
    string tmp;
    string total = "";
    while (getline(input, tmp)) {
        total += tmp + '\n';
    }
    input.close();

    // Parse file
    cout << total << endl; 
    BusinessCardParser bcp = BusinessCardParser();
    ContactInfo ci = bcp.getContactInfo(total);

    cout << "==>\n\n";

    cout << "Name: " << ci.getName() << endl;
    cout << "Phone: " << ci.getPhoneNumber() << endl;
    cout << "Email: " << ci.getEmailAddress() << endl;

    return 0;
}

File: driver.py
import sys
from BusinessCardParser import BusinessCardParser
from ContactInfo import ContactInfo

def main():
    if len(sys.argv) != 2:
        print("usage:", sys.argv[0], "<filename>")
        return 1
    # read from file
    input = open(sys.argv[1], "r")
    total = input.read()
    print(total)

    # parse file
    bcp = BusinessCardParser()
    ci = bcp.getContactInfo(total)

    print("==>\n")
    
    print("Name:", ci.getName)
    print("Phone:", ci.getPhoneNumber)
    print("Email:", ci.getEmailAddress)
main()